iceoryx_examples/request_response_in_c/client_c_basic.c

```diff
@@ -0,0 +1,128 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_binding_c/client.h"
+#include "iceoryx_binding_c/request_header.h"
+#include "iceoryx_binding_c/response_header.h"
+#include "iceoryx_binding_c/runtime.h"
+#include "request_and_response_c_types.h"
+#include "sleep_for.h"
+
+#include <signal.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdio.h>
+
+bool keepRunning = true;
+const char APP_NAME[] = "iox-c-request-response-client-basic";
+
+void sigHandler(int signalValue)
+{
+    (void)signalValue;
+    keepRunning = false;
+}
+
+int main()
+{
+    //! [register signal handler and init runtime]
+    signal(SIGINT, sigHandler);
+    signal(SIGTERM, sigHandler);
+
+    iox_runtime_init(APP_NAME);
+    //! [register signal handler and init runtime]
+
+    //! [create client]
+    iox_client_storage_t clientStorage;
+    iox_client_t client = iox_client_init(&clientStorage, "Example", "Request-Response", "Add", NULL);
+    //! [create client]
+
+    //! [define variables]
+    uint64_t fibonacciLast = 0U;
+    uint64_t fibonacciCurrent = 1U;
+    int64_t requestSequenceId = 0;
+    int64_t expectedResponseSequenceId = requestSequenceId;
+    //! [define variables]
+
+    //! [main loop]
+    while (keepRunning)
+    {
+        //! [loan request]
+        struct AddRequest* request = NULL;
+        enum iox_AllocationResult loanResult =
+            iox_client_loan_request(client, (void**)&request, sizeof(struct AddRequest));
+        //! [loan request]
+
+        if (loanResult == AllocationResult_SUCCESS)
+        {
+            //! [set sequence id]
+            iox_request_header_t requestHeader = iox_request_header_from_payload(request);
+            iox_request_header_set_sequence_id(requestHeader, requestSequenceId);
+            expectedResponseSequenceId = requestSequenceId;
+            requestSequenceId += 1;
+            //! [set sequence id]
+
+            //! [set and send request]
+            request->augend = fibonacciLast;
+            request->addend = fibonacciCurrent;
+            printf("%s Send Request: %lu + %lu\n",
+                   APP_NAME,
+                   (unsigned long)fibonacciLast,
+                   (unsigned long)fibonacciCurrent);
+            iox_client_send(client, request);
+            //! [set and send request]
+
+            //! [wait for response]
+            const uint32_t DELAY_TIME_IN_MS = 150U;
+            sleep_for(DELAY_TIME_IN_MS);
+            //! [wait for response]
+
+            //! [process response]
+            const struct AddResponse* response = NULL;
+            while (iox_client_take_response(client, (const void**)&response) == ChunkReceiveResult_SUCCESS)
```

Comment:
We only expect one response as far as I understand. Why the `while`? Of course it does no harm but, what if we did not wait long enough and no response is available yet?

I think we should sleep and then spin until we get our response. It would be better not to spin but also a little more complex with e.g. waitset which is covered in another example.

Reply:
I would leave it as it is. For two reasons:
* The cpp example has the identical approach and I wanted to make sure that the C example is as close as possible to the C++ version
* You are right but most of the times our examples are used as templates for bigger projects and with this in mind I think it is the right approach - maybe one introduces some number cruncher somewhere and than one would start to miss responses.

Reply:
It might also be possible to not get a response in time and the next time the queue is polled there would be two responses

Reply:
@elBoberido But I think here we are in an error case (defined by the user, not in general). That is why I would like clarity whether we are waiting for the response indefinitely before sending the next request or just for some time. In the latter case this can happen and then we need the loop for sure.

I just want to understand the intended basic use-case and the intricacies that can occur.